File: reactapp/src/models/HR/EndMonthData.ts
import { CompanyEmployeeTax, CompanyEmployeeTaxInsert, createDefaultCompanyEmployeeTax, defaultCompanyEmployeeTaxUpdate } from "./CompanyEmployeeTax";
import { createDefaultIncomeInsert, defaultIncomeUpdate, Income, IncomeInsert } from "./Income";
import { createDefaultScheduleInsert, defaultScheduleUpdate, Schedule, ScheduleInsert } from "./Schedule";

export interface EndMonthDataInsert
{
    income: IncomeInsert
    schedule: ScheduleInsert
    companyEmployeeTax: CompanyEmployeeTaxInsert
}

export const createEndMonthDataInsert = (employeeId: int): EndMonthDataInsert => {
    return {
        income: createDefaultIncomeInsert(employeeId),
        schedule: createDefaultScheduleInsert(employeeId),
        companyEmployeeTax: createDefaultCompanyEmployeeTax(employeeId)
    }
}


export interface EndMonthDataUpdate
{
    income: Income
    schedule: Schedule
    companyEmployeeTax: CompanyEmployeeTax
}

export const createEndMonthDataUpdate = (): EndMonthDataUpdate => {
    return {
        income: defaultIncomeUpdate,
        schedule: defaultScheduleUpdate,
        companyEmployeeTax: defaultCompanyEmployeeTaxUpdate
    }
}

File: webapi/Models/HR/EndMonthData.cs
namespace webapi.Models.HR
{
    public record EndMonthDataInsert
    {
        public required IncomeDTO Income { get; set; }
        public required ScheduleDTO Schedule { get; set; }
        public required CompanyEmployeeTaxDTO CompanyEmployeeTax { get; set; }
    }
    public record EndMonthDataUpdate
    {
        public required Income Income { get; set; }
        public required Schedule Schedule { get; set; }
        public required CompanyEmployeeTax CompanyEmployeeTax { get; set; }
    }
}
